core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryParser.java

```diff
@@ -49,31 +45,41 @@ public MultiMatchQueryParser() {
     }
 
     @Override
-    public Query parse(QueryShardContext context) throws IOException, QueryParsingException {
-        QueryParseContext parseContext = context.parseContext();
+    public MultiMatchQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
         Object value = null;
-        float boost = AbstractQueryBuilder.DEFAULT_BOOST;
-        Float tieBreaker = null;
-        MultiMatchQueryBuilder.Type type = null;
-        MultiMatchQuery multiMatchQuery = new MultiMatchQuery(context);
+        Map<String, Float> fieldsBoosts = new HashMap<>();
```

Comment:
The key, values will be put into the builder Treemap, so it really does not matter as long as it is a Map.